### ISSUE TITLE
Running Consumer Route of MDX Fails on Vite Plugin version of Remix

### ISSUE THREAD

**Issue by bronifty** [remix-run/remix#7901]
### What version of Remix are you using?

2.2.0

### Are all your remix dependencies & dev-dependencies using the same version?

- [X] Yes

### Steps to Reproduce

use remix as a vite plugin, build an app with a tsx route that consumes mdx via the @mdx-js/rollup plugin and build. head to the mdx consumer route and it breaks with the error message 
TypeError: t.jsxDEV is not a function
    at o (http://localhost:3000/build/assets/test-1a250893.js:9:223)
    at l (http://localhost:3000/build/assets/test-1a250893.js:10:578)
    at io (http://localhost:3000/build/assets/entry.client-1dd82fdc.js:68:19477)
    at ja (http://localhost:3000/build/assets/entry.client-1dd82fdc.js:70:43691)
    at Da (http://localhost:3000/build/assets/entry.client-1dd82fdc.js:70:39496)
    at id (http://localhost:3000/build/assets/entry.client-1dd82fdc.js:70:39427)
    at $r (http://localhost:3000/build/assets/entry.client-1dd82fdc.js:70:39287)
    at Ni (http://localhost:3000/build/assets/entry.client-1dd82fdc.js:70:35709)
    at Ta (http://localhost:3000/build/assets/entry.client-1dd82fdc.js:70:34665)
    at E (http://localhost:3000/build/assets/entry.client-1dd82fdc.js:55:1562)
    
    
    
    

### Expected Behavior

i expect it to not break

### Actual Behavior

it breaks. here's the repo to repro: https://github.com/bronifty/migrate-remix-to-vite.git

**Comment by hi-ogawa** [comment]
It looks like mdx plugin is transpiling JSX into `jsxDEV(...)` instead of `jsx(...)` even for build command.
I tested your reproduction and setting mdx plugin's `development` option explicitly seems to solve the issue:

```tsx
export default defineConfig((env) => ({
  plugins: [
   ...
    mdx({
      development: env.command === "serve",  /// <==== switch based on `vite dev` or `vite build`
      remarkPlugins: [
        remarkFrontmatter,
        remarkMdxFrontmatter,
      ],
    })
  ],
}));
```

If this is not really handled by `mdx` plugin itself, maybe it's worth mentioning in remix doc https://remix.run/docs/en/main/future/vite#add-mdx-plugin

**Comment by bronifty** [comment]
it was tough following this thread but I checked the file you changed in the PR and it works thanks

**Comment by hi-ogawa** [comment]
Sorry about my comment derailing a bit too much...
Thanks for testing and providing the reproduction!

**Comment by bronifty** [comment]
no problem thank you!

**Comment by markdalgleish** [comment]
Fixed by #7911.
